lib/puppet/functions/eyaml_lookup_key.rb

```diff
@@ -44,7 +44,7 @@ def load_data_hash(options, context)
     path = options['path']
     context.cached_file_data(path) do |content|
       begin
-        data = YAML.load(content, path)
+        data = Puppet::Util::Yaml.safe_load(content, [Symbol], path)
```

Comment:
More of ruby q: At runtime does Symbol translate into the class name?

Reply:
It's the Symbol class:

```
irb(main):003:0> Symbol.is_a?(Class)
=> true
irb(main):004:0> Symbol.name
=> "Symbol"
```